source/ID/concordium-did.rst

```diff
@@ -0,0 +1,661 @@
+
+From the command line, ``concordium-client`` allows for retrieving the data in the following way:
+
+.. code-block:: console
+
+  $concordium-client contract show IND
+
+.. TODO update, once we have a DID resolver
+
+
+.. _sci-reference:
+
+Smart Contract Instance Reference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+*Dereferencing* the smart contract DID reference invokes the specified entrypoint.
+
+Dereferencing a DID reference of the form
+
+``did:ccd:NET:sci:IND:SUBIND/EP[/PAR]``
+
+can be done by using the gRPC interface command ``InvokeInstance``.
+The entrypoint is considered a *view*: no state changes are persisted, only the result of the invocation is returned to the caller.
+The parameter ``PAR`` is passed to the entrypoint.
+
+The result of the invocation is the return value produced by the contract or an error, if the invocation failed.
+
+If the contract contains an embedded schema, then the response is the following:
+
+.. code-block:: json
+
+  {
+    "type" : "json",
+    "data": "JSON"
+  }
+
+Where ``JSON`` is the JSON rendering of the response.
+
+If the contract does not contain an embedded schema, then the response is the following:
+
+.. code-block:: json
+
+  {
+    "type" : "raw",
+    "data": "BASE16DATA"
+  }
+
+Where ``BASE16DATA`` is a base16-encoded return value.
+
+From the command line, ``concordium-client`` allows for invoking a smart contract instance in the following way:
+
+.. code-block::
+
+  $concordium-client contract invoke IND --entrypoint EP --parameter-binary param.bin
+
+The base16 encoding of the ``param.bin`` file corresponds to ``PAR``.
+
+See the details in the `gRPC v2 documentation`_.
+
+.. seealso::
+
+  `Dereferencing a DID URL`_ in the W3C Credentials Community Group draft report.
+
+.. TODO update, once we have a DID resolver
+
+Public Key Cryptography DID
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The DID document corresponding to a DID of the form
+
+``did:ccd:NET:pkc:PK``
+
+can be constructed directly from the DID without any lookup necessary.
+
+.. note::
+
+  The ``NET`` part is optional and currently there is no difference how the documents are generated for different networks.
+  In the future, however, the ``vefiricationMethod`` as it specified in :ref:`concordium-did-pkc` might depend on the network.
+
+Identity Provider DID
+^^^^^^^^^^^^^^^^^^^^^
+
+The DID document information for a DID of the form
+
+``did:ccd:NET:idp:INDEX``
+
+can be resolved by looking up an identity provider ``INDEX`` on blockchain ``NET``.
+
+Data required to construct the DID document can be acquired by using the gRPC interface command ``GetIdentityProviders``.
+
+See the details in the `gRPC v2 documentation`_.
+
+From the command line, ``concordium-client`` allows to retrieve the data in the following way:
+
+.. code-block:: console
+
+    $concordium-client raw GetIdentityProviders
+
+
+Update
+------
+
+At this time Concordium does not support the update of DID documents.
```

Comment:
Credential and account keys can be updated, not sure if that matters.

Reply:
I added some general info referencing Rust code documentation. Here and below for deactivation.